### ISSUE TITLE
Trim data to test training stages

### ISSUE THREAD

**Issue by aimalz** [LSSTDESC/rail_attic#200]
The tests take a really long time to run. Let's just use a smaller amount of data to test the super-slow nontrivial training stages.

**Comment by sschmidt23** [comment]
I believe the test data in question is 100 galaxies to train and 10 galaxies to test, it's hard to get much smaller.

**Comment by sschmidt23** [comment]
You can run pytest locally and see which tests are taking the bulk of the time, if I remember it's the testing of PZFlowPDF and delightPZ, particularly delightPZ, which is just super slow.

**Comment by jfcrenshaw** [comment]
Hmm when I ran them earlier, it seemed to sit on the FlexZBoost test for 90% of the time

**Comment by sschmidt23** [comment]
Oh, yes, if you don't specify xgboost==0.90.0 (which you can no longer do for python3.9) then for some reason the training of the xgboost regressor takes way longer in newer versions of xgboost.  I tried digging into this and did not see what about xgboost has changed, so I don't know why it is so much slower.  However, I didn't think it was *that* much slower, I thought it was till Delight that dominated the time.

**Comment by sschmidt23** [comment]
It is possible that recent updates to xgboost have made the regressor training *even slower*, in which case what I said could be wrong.  Maybe it's time to ping Ann Lee et al. to ask if they have any updates to the FlexCode package that could speed it back up?

**Comment by jfcrenshaw** [comment]
I might have also been wrong about which test it was. I'll make sure I confirm next time I run the tests.

Edit: confirmed

**Comment by sschmidt23** [comment]
Hmm, I don't know what's going on: I pulled out just the flexzboost test from test_algos, on my home machine it takes 18.59 seconds to run with python3.9 and .  I set it up on NERSC, with python3.8 and xgboost==0.90.0 (which I thought was the problem) takes 2169.5 seconds, more than half an hour! With python3.9 and xgboost==1.6.0 I killed it after an hour when it still wasn't done.
Obviously the ubuntu version run by GH Action is somewhere in between, since the sum total of tests takes ~10 minutes.

So, something weird is going on with FlexCode with different setups.
